lib/dry/configurable/setting.rb

```diff
@@ -66,7 +66,7 @@ def initialize(name, input: Undefined, default: Undefined, **options)
         @default = default
         @options = options
 
-        evaluate if input_defined?
+        value if input_defined?
```

Comment:
I can see why you've removed the private `#evaluate` method here, to keep all the logic in `#value` itself, but the net result is that this line here reads a lot worse – it's just a random noun hanging out there on its own line, rather than a verb describing the outcome of the method (which, in this case, is actually about a side effect, so a verb makes sense).

Is this just a change you just made along the way because you thought it would be nice, or is it critical to your work in this PR?

Reply:
`@value ||=` instead of `evaluated?` leads to evaluating value during `cloneable?` check from `initialize_copy` which fails if `finalize!(freeze_values: true` is called. I don't quite like how it looks either, added alias evaluate -> value.